### PR TITLE
Set data storage location

### PR DESCRIPTION
**Pull request by joycelynng**
Included SetStorageCommandTest

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/SetStorageCommand.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        	
+        	Path oldStorageFileFilePath = filePaths.getValue();
+        	
+        	model.setStorage(newStorageFileFilePath.toFile(), oldStorageFileFilePath.toFile()); //Throws IOException
+        	
+        	return new CommandResult(String.format(MESSAGE_SUCCESS, model.getTaskManagerStorageFilePath()));
+        	
+        } catch (InvalidPathException ex){
+        	model.loadPreviousState();
+        	return new CommandResult(String.format(MESSAGE_INVALID_PATH_EXCEPTION));
```
Comment (kabir-plod) [inline src/main/java/seedu/address/logic/commands/SetStorageCommand.java]: Change model.loadPreviousState() to the new model.undoSaveState()

File: src/main/java/seedu/address/logic/parser/Parser.java [hunk 2]
```diff
@@ -55,8 +55,11 @@
 
 	private static final Pattern SOMEDAY_ARGS_FORMAT = Pattern.compile("'(?<taskName>.*\\S+.*)'");
 
+	//@@author A0143756Y
+	private static final Pattern SET_STORAGE_ARGS_FORMAT = Pattern.compile
+			("(?<folderFilePath>(\\s*[^\\s+])+)\\s+save-as\\s+(?<fileName>(\\s*[^\\s+])+)");
```
Comment (kabir-plod) [inline src/main/java/seedu/address/logic/parser/Parser.java]: `[^\\s+]` term will reject file paths with spaces in them. `.+` proabably better

File: src/main/java/seedu/address/model/ModelManager.java [hunk 3]
```diff
@@ -142,6 +157,52 @@ public synchronized void checkForOverdueTasks() {
...
+    }
+    
+    @Override
+    public synchronized void setStorage(File newStorageFile, File oldStorageFile) throws IOException{
+    	assert newStorageFile!= null;
+    	assert oldStorageFile!= null;
+    	assert !newStorageFile.equals(oldStorageFile);
+    	
+    	Files.copy(oldStorageFile, newStorageFile);  //Throws IOException
+
```
Comment (kabir-plod) [inline src/main/java/seedu/address/model/ModelManager.java]: Is the old file deleted?

**Comment by joycelynng** [comment]
Has been merged into debug-set-storage branch

**Pull request closed without merging**